### ISSUE TITLE
Replace some ReactDOMRe with equivalent ReactDOM apis

### ISSUE THREAD

**Issue by ryyppy** [davesnx/styled-ppx#164]
The ppx currently uses some APIs that are considered obsolete and have a proper equivalent in  `ReactDOM` / `React`:

**In the source code:**

- `ReactDOMRe.domRef` -> `ReactDOM.domRef` (see [this code](https://github.com/davesnx/styled-ppx/blob/9451b3725755406554bb2c353c7cc17c7d8bdae0/packages/ppx/src/styled_ppx.re#L370))

**In the docs:**

```rescript

ReactDOMRe.renderToElementWithId(
  <Dynamic color="#EB5757" background="#516CF0">
    {React.string("Hello!")}
  </Dynamic>,
  "app",
);
```

is equivalent to the modern version:

```rescript
switch(ReactDOM.querySelector("#app")){
  | Some(root) => ReactDOM.render(<Dynamic color="#EB5757" background="#516CF0">
    {React.string("Hello!")}
  </Dynamic>, root)
  | None => () // do nothing
}
```

These changes should actually not break any code that is based on the latest `reason-react` version. It will also make it possible to migrate over to `@rescript/react` later on

**Comment by davesnx** [comment]
Hey @ryyppy, Thanks for letting me know.

I wasn't aware of those deprecations, will release a fix with that.

**Comment by davesnx** [comment]
This fix landed in master https://github.com/davesnx/styled-ppx/issues/164, are available on the 1.3.x nightly release and soon as a stable version.

Thanks for reporting it @ryyppy
